scrape_data/cta_data_downloads.py

```diff
@@ -60,56 +60,166 @@ def save_csv_to_bucket(df: pd.DataFrame, filename: str) -> None:
         .put(Body=csv_buffer.getvalue())
 
 
-def save_sched_daily_summary() -> None:
-    data = sga.GTFSFeed.extract_data(CTA_GTFS)
-    data = sga.format_dates_hours(data)
-    trip_summary = sga.make_trip_summary(data)
-
-    route_daily_summary = (
-        sga.summarize_date_rt(trip_summary)
-    )
-    route_daily_summary['date'] = route_daily_summary['date'].astype(str)
-    route_daily_summary_today = route_daily_summary.loc[route_daily_summary['date'] == today]
-
-    print(f'Saving cta_route_daily_summary_{today}.csv to public bucket')
-    filename = f'schedule_summaries/daily_job/cta_route_daily_summary_{today}.csv'
-    save_csv_to_bucket(
-        route_daily_summary_today,
-        filename=filename
+def save_sched_daily_summary(date_range: List[str] = None) -> None:
+    if date_range is None:
+        date_range = [today]
+        print(f"No date range given. Using {today} only")
+            
+    start_date = pendulum.parse(min(date_range))
+    end_date = pendulum.parse(max(date_range))
+    period = pendulum.period(start_date, end_date)
+    full_date_range = [dt.to_date_string() for dt in period.range('days')]
+    zip_filename_list = [f'cta_schedule_zipfiles_raw/google_transit_{date}.zip'
+                         for date in full_date_range]
+
+    # Check for files in bucket.
+    found_list = keys(
+        csrt.BUCKET_PUBLIC,
+        zip_filename_list
     )
-    print(f'Confirm that {filename} exists in bucket')
-    keys(csrt.BUCKET_PUBLIC, [filename])
+    def confirm_saved_files(file_dict: dict) -> None:
+        for fname in ['csv_filenames', 'zip_filenames']:
+            print('Confirm that ' + ', '.join(file_dict[fname])
+                + ' exist in bucket')  
+            _ = keys(csrt.BUCKET_PUBLIC, file_dict[fname])
```

Comment:
The code in the `keys` function implicitly confirms that the file has been saved?
Is it true that `file_dict` is a mapping from a string to a list of strings?

Reply:
Yes, the `keys` function confirms that a list of filenames exists in the bucket, and `file_dict` has values that are lists of strings.